mbed_tools_ci_scripts/utils/git_helpers.py

```diff
@@ -590,8 +590,15 @@ def __enter__(self) -> ProjectGitWrapper:
         return self._repo
 
     def __exit__(self, type: Any, value: Any, traceback: Any) -> None:
-        """Context manager exit point."""
-        pass
+        """Context manager exit point.
+
+        As described in
+        https://github.com/gitpython-developers/GitPython/blob/60acfa5d8d454a7c968640a307772902d211f043/git/repo/base.py#L223,
+        Tempfiles objects on Windows are holding references to open files until
+        they are collected by the garbage collector, thus preventing deletion.
+        """
+        print("__exit__ called")
```

Comment:
Should we remove this `print` statement?

Reply:
Also looks like we've got test failures. If your previous fix with `__del__` was working lets revert back to it, this will all be going away soon anyway.

Reply:
> Should we remove this `print`?

Removed, added for debugging purpose.

Reply:
> Also looks like we've got test failures. If your previous fix with `__del__` was working lets revert back to it, this will all be going away soon anyway.

One last attempt to use `__exit__` to fix the issue. If this doesn't work then I'll revert to `__del__` fix.

Reply:
Finally CI is happy with `__exit__` fix.